### PR TITLE
Keyboard key

### PR DESCRIPTION
**Pull request by jamesadevine**

### CODE REVIEW

File: jm-keyboard-key-46-1.0/keyboard-key.prj [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Break6=No 0000 00FF
+Break7=No 0000 00FF
+Break8=No 0000 00FF
+[Ctrl_Break]
+Mark=51 1
+ICE=0 6 2 CE4
+[HEAD]
+[DEPEND]
+~$:INC_PDK\PMS150C.INC
+~C:\development\software\padauk\ms-jd-padauk\jd\jdheader.asm
```
Comment (mmoskal) [inline jm-keyboard-key-46-1.0/keyboard-key.prj]: make sure to use relative paths (IDE will overwrite them now and then - just don't check the absolute paths in...)

File: jm-keyboard-key-46-1.0/main.asm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// all pins on PA
+#define PIN_LED	5
+#define PIN_JACDAC 6
+//#define PIN_LOG 7
+
+// Cost given in comment: words of flash/bytes of RAM
+// #define CFG_T16_32BIT 1 // 1/1
+#define CFG_BROADCAST 1 // 20/0
+#define CFG_RESET_IN 1 // 24/1
+#define CFG_FW_ID 0x3f9ca24e // 24/0
```
Comment (mmoskal) [inline jm-keyboard-key-46-1.0/main.asm:22]: update f/w id

**Review by mmoskal — CHANGES_REQUESTED** [review]

**Comment by mmoskal** [comment]
update jm-button* fw-id in this PR?

**Comment by jamesadevine** [comment]
Done! Let me know if good to merge

File: jm-button-10-1.1/main.asm [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
+// all pins on PA
+#define PIN_LED	3
+#define PIN_JACDAC 6
+#define PIN_LOG 7
+
+// Cost given in comment: words of flash/bytes of RAM
+// #define CFG_T16_32BIT 1 // 1/1
+#define CFG_BROADCAST 1 // 20/0
+#define CFG_RESET_IN 1 // 24/1
+#define CFG_FW_ID 0x95b66a3f // 24/0
```
Comment (mmoskal) [inline jm-button-10-1.1/main.asm]: f/w ID has to start with 0x3

File: jm-button-10-1.2/main.asm [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+// all pins on PA
+#define PIN_LED	5
+#define PIN_JACDAC 6
+//#define PIN_LOG 7
+
+// Cost given in comment: words of flash/bytes of RAM
+// #define CFG_T16_32BIT 1 // 1/1
+#define CFG_BROADCAST 1 // 20/0
+#define CFG_RESET_IN 1 // 24/1
+#define CFG_FW_ID 0x5cafeda1 // 24/0
```
Comment (mmoskal) [inline jm-button-10-1.2/main.asm]: 0x3

**Review by mmoskal — CHANGES_REQUESTED** [review]
we have a range reserved for f/w id

**Comment by jamesadevine** [comment]
Ok will fix up!

James
________________________________

File: jm-button-10-1.2/main.asm (hunk 4 above)
Comment (jamesadevine) [reply jm-button-10-1.2/main.asm]: Patched!

**Review by mmoskal — APPROVED** [review]

**Pull request merged**